apps/framework-cli/src/infrastructure/processes/cron_registry.rs

```diff
@@ -27,30 +48,99 @@ impl CronRegistry {
             Box::pin(async move {
                 let _ = task();
             })
-        })?;
+        })
+        .map_err(|e| anyhow!(e))?;
 
         let scheduler = self.scheduler.lock().await;
-        scheduler.add(job).await?;
+        scheduler.add(job).await.map_err(|e| anyhow!(e))?;
 
         Ok(())
     }
 
-    pub async fn start(&self) -> Result<(), JobSchedulerError> {
-        let scheduler: Arc<Mutex<JobScheduler>> = self.scheduler.clone();
-        task::spawn(async move {
-            let scheduler = scheduler.lock().await;
-            scheduler.start().await?;
+    pub async fn start(&self) -> Result<()> {
+        let scheduler = self.scheduler.lock().await;
+        scheduler.start().await.map_err(|e| anyhow!(e))?;
+        Ok(())
+    }
 
-            Ok::<(), JobSchedulerError>(())
-        });
+    pub async fn stop(&self) -> Result<()> {
+        let mut scheduler = self.scheduler.lock().await;
+        scheduler.shutdown().await.map_err(|e| anyhow!(e))?;
 
         Ok(())
     }
 
-    pub async fn stop(&self) -> Result<(), JobSchedulerError> {
-        let mut scheduler = self.scheduler.lock().await;
-        scheduler.shutdown().await?;
+    pub async fn register_jobs(&self, project: &Project) -> Result<()> {
+        info!("<cron> Registering cron jobs from project configuration");
+        info!("<cron> Cron jobs: {:?}", project.cron_jobs);
+
+        for job in &project.cron_jobs {
+            let job_id = job.job_id.clone();
+            let cron_spec = job.cron_spec.clone();
+            let script_path = job.script_path.clone();
+            let url = job.url.clone();
+
+            // Ensure at least one of script_path or url is present
+            if script_path.is_none() && url.is_none() {
+                return Err(anyhow!(
+                    "At least one of script_path or url must be present for job: {}",
+                    job_id
+                ));
+            }
+
+            self.add_job(&cron_spec, move || {
+                if let Some(ref path) = script_path {
+                    // Execute the script based on file extension
+                    let extension = Path::new(path)
+                        .extension()
+                        .and_then(std::ffi::OsStr::to_str)
+                        .unwrap_or("");
+
+                    let output = match extension {
+                        "js" => Command::new("node").arg(path).output(),
+                        "ts" => Command::new("moose-exec").arg(path).output(),
```

Comment:
Now that you are using this - you will probably want to add as well the .bin to the path: https://github.com/514-labs/moose/blob/main/apps/framework-cli/src/framework/typescript/bin.rs#L26-L35